### ISSUE TITLE
Problem: can't find the most interesting Nostr events

### ISSUE THREAD

**Issue by stanstacks** [Stackerstan/mindmachine#2]
Problem: can't find the most interesting Nostr events

There's a lot of stuff happening on Nostr, and it's hard to find the most interesting stuff.

Possible Solution:
A stackerstan Mind that implements the [Nostranker protocol](https://stackerstan.org/index.html#nostranker).

It would fetch all events that have human readable content (regex the content field) and make these events available to be scored by `Kind 641402` rankings.

The frontend would display a user's feed as usual (events from people they follow) but would also mix in random events from people they do not follow.  

When they click "upvote" they will be asked "upvote this more than what?" and be shown a list of events they have previously ranked (if any) or random events from their follow list if not. 

They will be asked to register on stackerstan and join the identity tree if they want their rankings to count globally, but anyone could request the Mind to combine any list of pubkey's rankings regardless of whether the pubkey is in the identity tree or not.

The default page for new users who don't follow anyone is just a list of events ranked by score.

**Comment by stanstacks** [comment]
The simplest architecture for this and other "heavy" use cases is probably a separate state machine (a sidemind?) that consumes state from the mindmachine (subscribes to events like the frontend) and simply sends an event containing its current state upon request (once a block is probably enough).

This mind would be heavy because it needs to index all nostr events.

The mindmachine can control the sidemind's message timing and order (required to get the same state at the same time). 

So to catch up, the mindmachine sends the sidemind an eventpack with all the events it needs to reproduce the current state. 

In live mode, the mindmachine will forward any event kinds (perhaps with filtering) that matches whatever the sidemachine asks for.

In both cases, this would just be another layer behind the Conductor and the Eventcatcher, retaining the existing turn-based message timing.

So basically exactly the same as the current architecture, except that instead of passing messages internally within the binary we are passing them between binaries using Nostr events.

**Comment by stanstacks** [comment]
A way for this to generate revenue:

Anyone can of course run their own Mindmachine and query locally for free. However not everyone will want to. 

Participants who run a public facing Mindmachine can claim an Expense for their time and the cost of the VPS if this solves a valid Problem (e.g.. it's painful to use unless you run a local mindmachine). Their implementation MUST follow the rules for the Expense to be valid.

One of those rules could be about giving priority to specific types of subscriptions, e.g. a company may want to sponsor certain types of subscription requests, but keep these protocol rules in mind:
* [**Rule**: Any revenue derived from Participants MUST be explicit and voluntary](https://stackerstan.org/index.html?protocol_id=4dffef504b1e7e076ba0fc25d204304d1fc5ec98fc68394047c1b6da49506b9f)
* [**Rule**: Revenue MUST NOT be derived by virtue of Participant attention](https://stackerstan.org/index.html?protocol_id=a96c1cf8db5c94e7e59eef19d90d33e3ed741750f643cbcc13700219bd8255f7)

It could also prioritize subscriptions from a client that has signed a kind 64xxxx event and sent it to the Mind (nostr subscription requests are not signed, so this is just a roundabout way to validate the client's pubkey and query the Mindmachine to see if this pubkey is on a list of "priority accounts").

So from there, the sky is the limit in terms of how you make a pubkey into a priority account - bidding based? Anyway there's an avenue to revenue, so this is a valid problem to solve.

**Comment by stanstacks** [comment]
Note to self:
Display events in order of their Nostranker score, but when they have the same score, sort them by how many mentions they have.
